apis/v1alpha2/config.go

```diff
@@ -69,27 +69,45 @@ func (c *AnyConfig) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Object)
 }
 
+// UnmarshalYAML implements an alternative parser this field.
```

Comment:
can you add a static type check for AnyConfig that it adheres to the YAML marshal/unmarshal interface?

Reply:
It is now based on go-yaml version 3. I have adapted the unmarshal signature accordingly and added a static type check.